src/Marten/Events/Projections/Aggregator.cs

```diff
@@ -20,9 +20,19 @@ public Aggregator()
                 .Where(x => x.Name == ApplyMethod && x.GetParameters().Length == 1)
                 .Each(method =>
                 {
+                    object step = null;
                     var eventType = method.GetParameters().Single<ParameterInfo>().ParameterType;
-                    var step = typeof (AggregationStep<,>)
-                        .CloseAndBuildAs<object>(method, typeof (T), eventType);
+                    if (eventType.GetTypeInfo().IsGenericType && eventType.GetGenericTypeDefinition() == typeof(Event<>))
```

Comment:
If you want to shortcut that code, there's an extension method in Baseline called `Closes(type)` that would do the same logic.